backlog-set-due-date

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh -e
+
+#redmine_api_key="${redmine_api_key:?"Need redmine API key"}"
+project_id="${project_id:-18}"
+ticket_limit="${ticket_limit:-200}"
+host="${host:-"https://progress.opensuse.org"}"
+issues=$(mktemp)
+issue=$(mktemp)
+
+#curl -s -H "X-Redmine-API-Key: $redmine_api_key" "$host/issues.json?project_id=$project_id&limit=$ticket_limit" >"$issues"
+
+cat ./issues.json | jq -r '.issues | .[] |  select(.priority.name!="Low") | select(.due_date==null) | select(.assignee==null)' >"$issues"
+
+for id in $(jq .id "$issues"); do
+  due_date=$(date -d"+14 days" +%Y-%m-%d)
+  echo "$id\t$due_date"
+  #TODO Setup the due date using the Progress API
```

Comment:
I remembered an older example where I created tickets in redmine automatically which can serve as a hopefully helpful example: https://progress.opensuse.org/projects/suseqa/wiki#create-Fate-tracker-tickets-from-Maritas-spreadsheet

Here I guess you need https://redmine.org/projects/redmine/wiki/Rest_Issues#Updating-an-issue

I tried that out right now, the following succeeded (with my api key instead of XXX):

```shell
curl -v -H 'X-Redmine-API-Key: XXX' -H 'Content-Type: application/json' -X PUT \
  -d '{"issue": {"due_date": "2020-11-30", "notes": "Setting due date based on mean cycle time of SUSE QE Tools"}}' \
  https://progress.opensuse.org/issues/71098.json
```

Reply:
Thanks!